packages/mobile/src/web3/saga.ts

```diff
@@ -167,47 +176,32 @@ export function* assignAccountFromPrivateKey(key: string) {
     } else {
       try {
         // @ts-ignore
-        account = yield call(web3.eth.personal.importRawKey, String(key), pincode)
+        yield call(web3.eth.personal.importRawKey, privateKey, pincode)
       } catch (e) {
         if (e.toString().includes('account already exists')) {
-          account = currentAccount
-          Logger.debug(
-            TAG + '@assignAccountFromPrivateKey',
-            'Importing same account as current one'
-          )
+          Logger.warn(TAG + '@assignAccountFromPrivateKey', 'Attempted to import same account')
         } else {
           Logger.error(TAG + '@assignAccountFromPrivateKey', 'Error importing raw key')
           throw e
         }
       }
       yield call(web3.eth.personal.unlockAccount, account, pincode, UNLOCK_DURATION)
+      web3.eth.defaultAccount = account
```

Comment:
Do you know why this was previously done only when `!zeroSyncMode`?

Reply:
It's still only done in !zerosyncmode, I just improved the organization.
It's think it's because we don't use web3 to sign txs in zero sync mode, not 100% sure tho

Reply:
ok 👍